pallets/loans/src/lib.rs

```diff
@@ -509,19 +513,34 @@ pub mod pallet {
 		/// Loan is accrued, NAV is update accordingly, and updates the LoanInfo with new write off index.
 		/// Cannot update a loan that was written off by admin.
 		/// Cannot write off a healthy loan or loan type that do not have maturity date.
-		#[pallet::weight(<T as Config>::WeightInfo::write_off())]
+		///
+		///
+		/// Weight is calculated for one group. Since there is no extra read or writes for groups more than 1,
+		/// We need to ensure we are charging the reads and write only once but the actual compute to be equal to number of groups processed
+		#[pallet::weight(Pallet::<T>::write_off_group_weight(T::MaxWriteOffGroups::get() as u64))]
 		pub fn write_off(
 			origin: OriginFor<T>,
 			pool_id: PoolIdOf<T>,
 			loan_id: T::LoanId,
-		) -> DispatchResult {
+		) -> DispatchResultWithPostInfo {
 			// ensure this is a signed call
 			ensure_signed(origin)?;
 
 			// try to write off
 			let index = Self::write_off_loan(pool_id, loan_id, None)?;
 			Self::deposit_event(Event::<T>::WrittenOff(pool_id, loan_id, index));
-			Ok(())
+
+			// since the write off group index is picked in loop sequentially,
+			// total loops = index+1
+
+			// this is soft limit, so runtime do not have enough in the deposit.
+			// charge the maximum we have deposited
+			let count = index + 1;
+			if count >= T::MaxWriteOffGroups::get() {
+				return Ok(().into());
+			}
+
+			Ok(Some(Self::write_off_group_weight(count as u64)).into())
```

Comment:
There is no way to determine the number of write-off-groups ahead?

Reply:
not without a storage call unfortunately